server/src/main/java/io/druid/server/coordinator/DruidCluster.java

```diff
@@ -113,9 +115,11 @@ private void addHistorical(ServerHolder serverHolder)
 
   public Collection<ServerHolder> getAllServers()
   {
-    return historicals.values().stream()
-                      .flatMap(Collection::stream)
-                      .collect(() -> realtimes, Set::add, Set::addAll);
+    final List<ServerHolder> allServers = historicals.values().stream()
```

Comment:
Ah, I think it would be ok. Actually ```DruidCluster``` is a snapshot of the ```ServerInventoryView```. ```DruidCoordinator``` builds a new ```DruidCluster``` at each run and this snapshot is used by subsequent ```DruidCoordinatorHelper```s in the same run.